### PR TITLE
[Doc] Add docker version description

### PR DESCRIPTION
**Pull request by xuhaiL**
## Why I'm doing:

If the docker version is less than 20.10.10, there are problems in constructing starrocks, and the docker version is not restricted in the document
[1] https://stackoverflow.com/questions/72841549/container-fails-to-start-insufficient-memory-for-the-java-runtime-environment-t
[2] https://github.com/adoptium/containers/issues/215

## What I'm doing:
fix this doc
Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [x] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.4
  - [x] 3.3
  - [x] 3.2
  - [x] 3.1
  - [x] 3.0
  - [x] 2.5

### CODE REVIEW

**Comment by alvin-celerdata** [comment]
@xuhaiL could you sign-off the CL by `git commit -s`?

File: docs/en/developers/build-starrocks/Build_in_docker.md [hunk 1]
```diff
@@ -45,7 +45,7 @@ Before compiling StarRocks, make sure the following requirements are satisfied:
 - **Software**
 
   - Your machine must be running on Ubuntu 22.04 or CentOS 7.9.
-  - You must have Docker installed on your machine.
+  - You must have Docker installed on your machine and version v20.10.1 at least.
```
Comment (kevincai) [inline docs/en/developers/build-starrocks/Build_in_docker.md:48]: version inconsistent with chinese doc.

**Comment by kevincai** [comment]
@xuhaiL didn't experience this issue in an older docker engine, can you elaborate a little bit on how to repro this?

**Comment by xuhaiL** [comment]
> @xuhaiL didn't experience this issue in an older docker engine, can you elaborate a little bit on how to repro this?

look this https://forum.mirrorship.cn/t/topic/16191


`
docker --version
Docker version 20.10.9, build c2ea9bc

docker pull starrocks/dev-env-ubuntu:2.5.22
imageId=4d030159caf7

docker run -it --name sr starrocks/dev-env-ubuntu:2.5.22

root@008675b94e82:~/starrocks# java -version
[0.003s][warning][os,thread] Failed to start thread "GC Thread#0" - pthread_create failed (EPERM) for attributes: stacksize: 1024k, guardsize: 4k, detached.
#
# There is insufficient memory for the Java Runtime Environment to continue.
# Cannot create worker GC thread. Out of system resources.
# An error report file with more information is saved as:
# /root/starrocks/hs_err_pid636.log

`

File: docs/en/developers/build-starrocks/Build_in_docker.md (hunk 1 above)
Comment (xuhaiL) [reply docs/en/developers/build-starrocks/Build_in_docker.md:48]: ok, let me fix it

**Comment by xuhaiL** [comment]
> @xuhaiL could you sign-off the CL by `git commit -s`?

i'm fix

**Review by kevincai — APPROVED** [review]

**Review by alvin-phoenix-ai — APPROVED** [review]

**Pull request merged**